crates/router/src/connector/nmi/transformers.rs

```diff
@@ -279,13 +279,13 @@ impl
             Response::Approved => (
                 Ok(types::PaymentsResponseData::TransactionResponse {
                     resource_id: types::ResponseId::ConnectorTransactionId(
-                        item.response.transactionid,
+                        item.response.transactionid.clone(),
```

Comment:
Add variable `orderid` in struct `StandardResponse` and use it as `connector_response_reference_id`
Refer https://secure.nmi.com/merchants/resources/integration/integration_portal.php#transaction_response_variables @TejasMate